### PR TITLE
Keeping track of last reload error for data sources (UA-542)

### PR DESCRIPTION
**Pull request by kotodharma**
I don't think we quite connected on what the UI for this should be, so I did what I was thinking, and can certainly change if it's not appropriate :=)

### CODE REVIEW

File: app/models/series.rb [hunk 1]
```diff
@@ -456,14 +456,14 @@ def new_transformation(name, data)
...
-    raise SeriesReloadException if update_spreadsheet.load_error?
+    raise SeriesReloadException, 'load error' if update_spreadsheet.load_error?
     #return self if update_spreadsheet.load_error?
 
     unless update_spreadsheet.class == UpdateCSV
       default_sheet = update_spreadsheet.sheets.first
       update_spreadsheet.default_sheet = sheet_to_load.nil? ? default_sheet : sheet_to_load
     end
-    raise SeriesReloadException unless update_spreadsheet.update_formatted?
+    raise SeriesReloadException, 'update not formatted' unless update_spreadsheet.update_formatted?
```
Comment (jonpage) [inline app/models/series.rb]: 👍

File: app/models/series.rb [hunk 2]
```diff
@@ -456,14 +456,14 @@ def new_transformation(name, data)
   def load_from(spreadsheet_path, sheet_to_load = nil)
     spreadsheet_path.gsub! ENV['DEFAULT_DATA_PATH'], ENV['DATA_PATH']
     update_spreadsheet = UpdateSpreadsheet.new_xls_or_csv(spreadsheet_path)
-    raise SeriesReloadException if update_spreadsheet.load_error?
+    raise SeriesReloadException, 'load error' if update_spreadsheet.load_error?
```
Comment (jonpage) [inline app/models/series.rb]: 👍

File: app/views/series/_source_list.html.erb [hunk 3]
```diff
@@ -33,6 +33,12 @@
 						%>&nbsp;<span style="color:gray">(<%= 
 						"#{ds.runtime.round(2)}s" unless ds.runtime.nil? 
 					%>)</span>
+        <% if ds.last_error %>
+          <p>
+            <span style='color:red;cursor:pointer;cursor:hand;' title='<%= ds.last_error %>'>Last error<br/>
+              <%= ds.last_error_at.localtime.strftime('%m/%d/%y %H:%M:%S') %>
+            </span>
```
Comment (jonpage) [inline app/views/series/_source_list.html.erb]: - [ ]  Closing paragraph tag

File: app/views/series/_source_list.html.erb [hunk 4]
```diff
@@ -33,6 +33,12 @@
 						%>&nbsp;<span style="color:gray">(<%= 
 						"#{ds.runtime.round(2)}s" unless ds.runtime.nil? 
 					%>)</span>
+        <% if ds.last_error %>
+          <p>
+            <span style='color:red;cursor:pointer;cursor:hand;' title='<%= ds.last_error %>'>Last error<br/>
```
Comment (jonpage) [inline app/views/series/_source_list.html.erb]: This UI is too sneaky. Let's directly output the error instead.

File: app/views/series/_source_list.html.erb [hunk 5]
```diff
@@ -35,9 +35,11 @@
 					%>)</span>
         <% if ds.last_error %>
           <p>
-            <span style='color:red;cursor:pointer;cursor:hand;' title='<%= ds.last_error %>'>Last error<br/>
+            <span style='color:red;' title='<%= ds.last_error %>'>Last error<br/>
               <%= ds.last_error_at.localtime.strftime('%m/%d/%y %H:%M:%S') %>
-            </span>
+            </span> <br/>
+            <%= ds.last_error %>
```
Comment (kotodharma) [inline app/views/series/_source_list.html.erb]: I recall that when we last discussed this it wasn't clear to me that we had decided exactly on a UI for the error, so I thought you might change what I did :=) Thx.

**Pull request merged**